rpi_backlight/utils.py

```diff
@@ -1,8 +1,39 @@
 from pathlib import Path
 from tempfile import TemporaryDirectory
+from typing import Optional
+
+from . import BoardType
 
 __all__ = ["FakeBacklightSysfs"]
 
+STRING_TO_BOARD_TYPE = {
+    "raspberry-pi": BoardType.RASPBERRY_PI,
+    "tinker-board": BoardType.TINKER_BOARD,
+    "tinker-board-2": BoardType.TINKER_BOARD_2,
+}
+
+BOARD_TYPE_TO_STRING = {
+    BoardType.RASPBERRY_PI: "raspberry-pi",
+    BoardType.TINKER_BOARD: "tinker-board",
+    BoardType.TINKER_BOARD_2: "tinker-board-2",
+}
+
+
+def detect_board_type() -> Optional[BoardType]:
+    try:
+        model_file = Path("/proc/device-tree/model")
+        model = model_file.read_text()
+    except OSError:
+        return None
+    if model.rfind("Tinker Board 2"):
+        return BoardType.TINKER_BOARD_2
+    elif model.rfind("Tinker Board"):
+        return BoardType.TINKER_BOARD
+    elif model.rfind("Raspberry Pi"):
+        return BoardType.RASPBERRY_PI
```

Comment:
```suggestion
    if model.startswith("Tinker Board 2"):
        return BoardType.TINKER_BOARD_2
    elif model.startswith("Tinker Board"):
        return BoardType.TINKER_BOARD
    elif model.startswith("Raspberry Pi"):
        return BoardType.RASPBERRY_PI
```

Reply:
Hi Linus,
mine says `ASUS Tinker Board 2` but the Tinker Board 1 says `Rockchip RK3288 Asus Tinker Board` its best we dont use startswith, as if it starts with ASUS or Rockchip it will not match.

Reply:
I see! that's fine then, but please add a comment mentioning exactly that.

Reply:
Comment added